torch/_ops.py

```diff
@@ -827,19 +827,43 @@ def load_library(self, path):
         ``torch.ops.loaded_libraries`` attribute, a set that may be inspected
         for the paths of all libraries loaded using this function.
 
+        It is possible for the custom operator library to specify Python
+        modules that they are associated with. If we are loading such a library,
+        then we will also import all of the associated Python modules.
+
         Args:
             path (str): A path to a shared library to load.
         """
         if torch._running_with_deploy():
             return
 
-        path = _utils_internal.resolve_library_path(path)
-        with dl_open_guard():
-            # Import the shared library into the process, thus running its
-            # static (global) initialization code in order to register custom
-            # operators with the JIT.
-            ctypes.CDLL(path)
-        self.loaded_libraries.add(path)
+        # The loaded libraries may be tied to python imports. If so, we want
+        # to retrieve the imports and then import them.
+        # See NOTE: [Requiring Python imports with TORCH_LIBRARY] for more details.
+        torch._C._dispatch_clear_required_pyimports()
+        try:
+            path = _utils_internal.resolve_library_path(path)
+            with dl_open_guard():
+                # Import the shared library into the process, thus running its
+                # static (global) initialization code in order to register custom
+                # operators with the JIT.
+                ctypes.CDLL(path)
+            self.loaded_libraries.add(path)
+            imports = torch._C._dispatch_unsafe_get_required_pyimports()
```

Comment:
Seems like you have a race here: in between the get here and the clear in finally, another thread could insert another pyimport onto the list which would be lost